src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/OperationInvokerBehavior.cs

```diff
@@ -12,8 +13,24 @@ public class OperationInvokerBehavior : IOperationBehavior
     {
         private IServiceProvider _serviceProvider;
 
+        private static readonly Lazy<bool> s_isOperationInvokerGeneratorEnabled = new Lazy<bool>(() =>
+        {
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+            foreach (var assembly in assemblies)
+            {
+                EnableCoreWCFOperationInvokerGeneratorAttribute attribute = assembly.GetCustomAttribute<EnableCoreWCFOperationInvokerGeneratorAttribute>();
```

Comment:
Use GetCustomAttributesData and not GetCustomAttribute as GetCustomAttribute instantiates an instance of the attribute. I'm also a little concerned about this approach of searching all assemblies currently loaded. You could reference a library which had this turned on and you aren't using any contracts from that library, and the app you compiled, you didn't turn it on. There would be no way to use that library without being forced to enable this in your main app. The only assembly you have definitely have control of is the main app assembly. What do you think about only paying attention to the attribute on main executing assembly? If everything is in a library, enable it there, but you also need to turn it on in your main app?

Reply:
I realized this won't work for xunit tests as the executing assembly will be the xunit test runner. We can't just hard code in something for xunit though as there are multiple different unit test frameworks. Need to give this some thought. The criteria is we need to work if there's a mix of assemblies where some have this enabled and some don't. What about not having a big global switch like this, but instead check the assembly for interface method. Could we have OperationDescription be populated with a bool saying whether to use a pre-generated invoker or not? You need to handle interface hierarchies, ie IService1 in Assembly1.dll was compiled with it, and IService2 derived from IService1 in Assembly2 was compiled without it. So methods IService1 have pregenerated invokers, but methods in IService2 don't. Or vice versa. Do you have any other ideas on how to handle this scenario?

Reply:
The code generation feature has to be enabled at service contract assembly level to generate code in the correct assembly.
However the usage of the generated code should be enabled at application host level. Would you be open to add an AppContext.Switch to force enabling usage of the generated code in unit tests ?

Reply:
That makes sense and keeps things simple. Go with the AppContext override.

Reply:
Great! PR is updated.